### PR TITLE
Patch for osx-arm64

### PR DESCRIPTION
**Pull request by FelixMoelder**
This PR includes a patch allowing ghostscript to be build for `osx-arm64`.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by FelixMoelder** [comment]
@conda-forge-admin, please rerender

**Comment by tdegeus** [comment]
@FelixMoelder In principle I'm open to this. Two questions:
* Isn't there some automatic migration?
* Are you hitting some serious obstacles?

**Comment by FelixMoelder** [comment]
It actually looks like there are some links missing for `osx-64` when applying that patch, while it is working locally for `osx-arm64`.
I expected this to be a simple patch without the need of messing around with the recipe.
As native support for `osx-arm64` is just around the corner (https://bugs.ghostscript.com/show_bug.cgi?id=703095#c2) I decided to leave this for now.

Edit:
There is automatic migration. But before that can be applied a recipe respectively the package itself must support that architecture which is why I intended to update the recipe first.

**Comment by tdegeus** [comment]
Let's wait for the native fix then! Thanks!

**Comment by FelixMoelder** [comment]
I'll give it a last try by only applying the patch for `osx-arm64`. ;)

**Comment by FelixMoelder** [comment]
It looks like the error is caused by rerendering and not by my changes.
I will therefore leave this PR open, as that issue needs to be solved anyway.
Has anyone a suggestion what needs to be done to get this running (@mbargull)?

File: recipe/build.sh [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
 
+if [[ $(uname) == Darwin ]]; then
+  export CC=clang
+  export CXX=clang++
+  export LDFLAGS="-L$PREFIX/lib -Wl,-rpath,$PREFIX/lib -headerpad_max_install_names $LDFLAGS"
+  export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
+  export MACOSX_DEPLOYMENT_TARGET="10.9"
+  export CXXFLAGS="-stdlib=libc++ $CXXFLAGS"
+fi
+
```
Comment (isuruf) [inline recipe/build.sh]: This is unneeded

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,10 +10,11 @@ source:
   url: https://github.com/ArtifexSoftware/ghostpdl-downloads/releases/download/gs{{ version | replace(".", "") }}/ghostscript-{{ version }}.tar.gz
   sha256: 6eaf422f26a81854a230b80fd18aaef7e8d94d661485bd2e97e695b9dce7bf7f
   patches:
+    - config.patch  # [arm64]
```
Comment (isuruf) [inline recipe/meta.yaml]: Please use the `gnuconfig` package to copy these as described in the blog post

Comment (FelixMoelder) [reply recipe/meta.yaml]: Thanks for pointing out the blog post. I was not aware of it.

Comment (FelixMoelder) [reply recipe/meta.yaml]: @isuruf For building on `osx-arm64` the `gnuconfig` package works indeed. Still, the CI is crashing on `osx-64`.
Any idea on how to solve this?

**Comment by FelixMoelder** [comment]
I will close this PR as it will be handled by https://github.com/conda-forge/ghostscript-feedstock/pull/17/files

**Pull request closed without merging**